monitoring/logs-api-options.html.md

```diff
@@ -0,0 +1,112 @@
+---
+title: Logs API options for programmatic access
+layout: docs
+nav: firecracker
+author: kcmartin
+date: 2025-10-01
+---
+
+## Overview
+
+Fly.io apps run on Firecracker microVMs we call Machines. Each Machine captures `stdout`/`stderr`, ships those logs over NATS, and stores them for a period of time in a Quickwit-backed search index. Most users consume logs via `fly logs` or by setting up log shipping to an external sink.
+
+But sometimes you want to grab logs directly, without a CLI or setting up an exporter, because you’re building a tool, automating something, or you just want to pipe logs into your own system.
+
+---
+
+## Options for fetching logs programmatically
+
+You can get logs programmatically via:
+
+- **Undocumented HTTP API**: What `fly logs` uses under the hood.
+- **NATS proxy**: Subscribe to real-time logs at the message level.
+- **Log shipper to external sink**: Ingest logs elsewhere, then query them via that system.
+
+Each has tradeoffs. If you want durable search, export logs. If you want low-latency push, use NATS. If you want to `curl` something now, use the HTTP API.
```

Comment:
I would say the main use case for the API is that it allows you fetch _historical_ logs up to the retention period (15 days or so). The other two options only start streaming logs from the moment you set them up, but they won't allow you to look at older logs.